src/lib/size.ts

```diff
@@ -0,0 +1,13 @@
+export function formatBytes (bytes: number, decimals = 0): string {
+  if (bytes === 0 || isNaN(bytes)) {
+    return '0 Bytes'
+  }
+
+  const k = 1024 // constant for the base of the exponentiation
+  const dm = decimals < 0 ? 0 : decimals // determines the number of decimal places to display
+  const sizes = ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'] // array of size units
```

Comment:
```
formatBytes(Number.MAX_SAFE_INTEGER) // '8 PiB'
```

we should eventually add support for bigint in this and update the type of `size` in file.tsx.